rust/connlib/tunnel/src/client.rs

```diff
@@ -614,84 +548,21 @@ impl ClientState {
                     username: offer.credentials.username,
                     password: offer.credentials.password,
                 },
-                domain,
+                domain: awaiting_connection.domain,
             },
         }));
     }
 
-    pub fn received_domain_parameters(
-        &mut self,
-        resource_id: ResourceId,
-        domain_response: DomainResponse,
-    ) -> connlib_shared::Result<()> {
-        let gateway_id = self
-            .gateway_by_resource(&resource_id)
-            .ok_or(Error::UnknownResource)?;
-
-        let peer_ips = self.dns_response(&resource_id, &domain_response, &gateway_id)?;
-
-        self.peers
-            .add_ips_with_resource(&gateway_id, &peer_ips, &resource_id);
-
-        Ok(())
-    }
-
-    fn dns_response(
-        &mut self,
-        resource_id: &ResourceId,
-        domain_response: &DomainResponse,
-        peer_id: &GatewayId,
-    ) -> connlib_shared::Result<Vec<IpNetwork>> {
-        let peer = self
-            .peers
-            .get_mut(peer_id)
-            .ok_or(Error::ControlProtocolError)?;
-
-        let resource_description = self
-            .resource_ids
-            .get(resource_id)
-            .ok_or(Error::UnknownResource)?
-            .clone();
-
-        let ResourceDescription::Dns(resource_description) = resource_description else {
-            // We should never get a domain_response for a CIDR resource!
-            return Err(Error::ControlProtocolError);
-        };
-
-        let resource_description =
-            DnsResource::from_description(&resource_description, domain_response.domain.clone());
-
-        let addrs: HashSet<_> = domain_response
-            .address
-            .iter()
-            .filter_map(|external_ip| {
-                peer.get_or_assign_translation(external_ip, &mut self.ip_provider)
-            })
-            .collect();
-
-        for addr in addrs.clone() {
-            self.dns_resources_internal_ips
-                .insert(addr, resource_description.clone());
-        }
-
-        send_dns_answer(self, Rtype::AAAA, &resource_description, &addrs);
-        send_dns_answer(self, Rtype::A, &resource_description, &addrs);
-
-        Ok(addrs.iter().copied().map(Into::into).collect())
-    }
-
     /// Attempt to handle the given packet as a DNS packet.
     ///
     /// Returns `Ok` if the packet is in fact a DNS query with an optional response to send back.
     /// Returns `Err` if the packet is not a DNS query.
     fn handle_dns<'a>(
         &mut self,
         packet: MutableIpPacket<'a>,
-        now: Instant,
     ) -> Result<Option<IpPacket<'a>>, (MutableIpPacket<'a>, IpAddr)> {
```

Comment:
The return type here is dated too, the caller should just call `.destination` IMO.